test/test_remap.py

```diff
@@ -142,11 +147,74 @@ def test_overwrite(self):
 
         # Perform remapping
         remap_to_edge_centers = source_grid['v1'].remap.nearest_neighbor(destination_grid=destination_dataset.uxgrid,
-                                                                                  remap_to="nodes")
+                                                                         remap_to="face centers", coord_type='cartesian')
 
         # Assert the remapped data is different from the original data
         assert not np.array_equal(destination_dataset['v1'], remap_to_edge_centers)
 
+    def test_source_data_remap(self):
+        """Test the remapping of all source data positions."""
+
+        # Open source and destination datasets to remap to
+        source_uxds = ux.open_dataset(mpasfile_QU, mpasfile_QU)
+        destination_dataset = ux.open_dataset(gridfile_geoflow, dsfile_v1_geoflow)
+
+        # Remap from `face_centers`
+        face_centers = source_uxds['latCell'].remap.nearest_neighbor(
+            destination_grid=destination_dataset.uxgrid,
+            remap_to="nodes"
+        )
+
+        # Remap from `nodes`
+        nodes = source_uxds['latVertex'].remap.nearest_neighbor(
+            destination_grid=destination_dataset.uxgrid,
+            remap_to="nodes"
+        )
+
+        # Remap from `edges`
+        edges = source_uxds['angleEdge'].remap.nearest_neighbor(
+            destination_grid=destination_dataset.uxgrid,
+            remap_to="nodes"
+        )
+
+        self.assertTrue(len(face_centers.values) != 0)
+        self.assertTrue(len(nodes.values) != 0)
+        self.assertTrue(len(edges.values) != 0)
+
+    def test_value_errors(self):
+        """Tests the raising of value errors and warnings in the function."""
+
+        # Open source and destination datasets to remap to
+        source_uxds = ux.open_dataset(gridfile_geoflow, dsfile_v1_geoflow)
+        source_uxds_2 = ux.open_dataset(mpasfile_QU, mpasfile_QU)
+        destination_dataset = ux.open_dataset(gridfile_geoflow, dsfile_v1_geoflow)
+
+        # Raise ValueError when `remap_to` is invalid
+        with nt.assert_raises(ValueError):
+            source_uxds['v1'].remap.nearest_neighbor(
+                destination_grid=destination_dataset.uxgrid,
+                remap_to="test", coord_type='spherical'
+            )
+        with nt.assert_raises(ValueError):
+            source_uxds['v1'].remap.nearest_neighbor(
+                destination_grid=destination_dataset.uxgrid,
+                remap_to="test", coord_type="cartesian"
+            )
+
```

Comment:
Switch from `ux.open_dataset` to `ux.open_grid` where possible,  replacing `destination_dataset.uxgrid` with `destination_grid` afterwards.